src/components/ScreenWrapper/BaseScreenWrapper.js

```diff
@@ -29,9 +29,18 @@ class BaseScreenWrapper extends React.Component {
     }
 
     componentDidMount() {
+        let willAlertModalBecomeVisible = false;
+
+        this.unsubscribeOnyx = onyxSubscribe({
+            key: ONYXKEYS.MODAL,
+            callback: (object) => {
+                willAlertModalBecomeVisible = object.willAlertModalBecomeVisible;
+            },
+        });
```

Comment:
This would be the first instance of such use of Onyx. What about moving this outside the component in lib and using the function to check the status of modal visibility? 

There are many examples of it in actions. e.g. https://github.com/Expensify/App/blob/9a13fa35726cf4a22542074276a96fba338c8234/src/libs/actions/Report.js#L31 

Because this is not an action, you can put it in a lib.

Reply:
Yeah i agree, that would be good, will reactor!